bundles/org.openhab.binding.cbus/src/main/java/org/openhab/binding/cbus/handler/CBusNetworkHandler.java

```diff
@@ -0,0 +1,274 @@
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+            } else if (network.isOnline()) {
+                logger.debug("Network is online");
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                logger.debug("Network is offline");
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Network is not reporting online");
+            }
+        } catch (CGateException e) {
+            logger.warn("Problem checking network state for network {}", network.getNetworkID(), e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+        }
+        if (!getThing().getStatus().equals(lastStatus)) {
+            ScheduledFuture<?> networkSync = getNetworkSync();
+            if (lastStatus == ThingStatus.OFFLINE) {
+                if (networkSync == null || networkSync.isCancelled())
+                    networkSync = scheduler.scheduleWithFixedDelay(networkSyncRunnable, (int) (10 * Math.random()),
+                            Integer.parseInt(getConfig().get(CBusBindingConstants.PROPERTY_NETWORK_SYNC).toString()),
+                            TimeUnit.SECONDS);
+            } else {
+                if (networkSync != null)
+                    networkSync.cancel(false);
+            }
+
+            for (Thing thing : getThing().getThings()) {
+                ThingHandler handler = thing.getHandler();
+                if (handler instanceof CBusGroupHandler) {
+                    ((CBusGroupHandler) handler).updateStatus();
+                }
+            }
+        }
+    }
+
+    public @Nullable synchronized CBusCGateHandler getCBusCGateHandler() {
+        if (this.bridgeHandler == null) {
+            logger.debug("getCBusCGateHandler --");
+            Bridge bridge = getBridge();
+            if (bridge == null) {
+                logger.warn("Required bridge not defined for device.");
+                return null;
+            }
+            ThingHandler handler = bridge.getHandler();
+            logger.debug("handler is {} ", handler);
+            if (handler instanceof CBusCGateHandler) {
+                this.bridgeHandler = (CBusCGateHandler) handler;
+            } else {
+                logger.debug("No available bridge handler found for bridge: {} .", bridge.getUID());
+                this.bridgeHandler = null;
+            }
+        }
+        return bridgeHandler;
+    }
+
+    public @Nullable Network getNetwork() {
+        return network;
+    }
+
+    public @Nullable ScheduledFuture<?> getInitNetwork() {
+        return initNetwork;
+    }
+
+    public @Nullable ScheduledFuture<?> getNetworkSync() {
+        return networkSync;
+    }
+
+    public @Nullable Project getProjectObject() {
+        return projectObject;
+    }
+
+    private Runnable networkSyncRunnable = new Runnable() {
```

Comment:
Make this a method and call it through lambdas

Reply:
Made method and called from Lambda